main.js

```diff
@@ -0,0 +1,312 @@
+            }
+          } else if (capsLock) {
+            if (element.getAttribute('valueRus') == null) {
+              addTextAtCaret(element.getAttribute('value'));
+              localStorage.setItem('result', result.value);
+            } else {
+              addTextAtCaret(element.getAttribute('valueRus').toUpperCase());
+            }
+            localStorage.setItem('result', result.value);
+          } else if (element.getAttribute('valueRus') == null) {
+            addTextAtCaret(element.getAttribute('value'));
+          } else {
+            addTextAtCaret(element.getAttribute('valueRus').toLowerCase());
+          }
+          break;
+      }
+      localStorage.setItem('result', result.value);
+    });
+  });
+};
+
+
+// Change language EN/RU with pressing 'AltLeft' and 'ShiftLeft'
+function runOnKeys(func, ...codes) {
+  const pressed = new Set();
+
+  document.addEventListener('keydown', (event) => {
+    document.getElementById('result').focus();
+    pressed.add(event.code);
+    if (event.code === 'CapsLock') {
+      upperLower();
+    }
+    if (document.getElementById(event.code) != null) {
+      document.getElementById(event.code).classList.remove('keyup');
+      document.getElementById(event.code).classList.add('keydown');
+    }
+
+    if (codes.some((code) => !pressed.has(code))) {
+      document.querySelectorAll('.key').forEach((element) => {
+        element.classList.remove('keyup');
+      });
+      return;
+    }
+
+    pressed.clear();
+    func();
+  });
+
+  // Clear highlight keys when out of input
+  document.getElementById('result').addEventListener('focusout', () => {
+    document.querySelectorAll('.key').forEach((element) => {
+      element.classList.remove('keydown');
+    });
+  });
+
+  document.addEventListener('keyup', (event) => {
+    try {
+      document.getElementById(event.code).classList.remove('keydown');
+      document.getElementById(event.code).classList.add('keyup');
+      pressed.delete(event.code);
+    } catch (error) {
+      console.warn(
+        `Нажатая клавиша "${event.key}" отсутствует на виртуальной клавиатуре`,
+      );
+    }
+  });
+}
+
+runOnKeys(
+  () => {
+    if (lang === 'en') {
+      lang = 'ru';
+    } else {
+      lang = 'en';
+    }
+    localStorage.lang = lang;
+    if (lang === 'en') {
+      document.querySelectorAll('span').forEach((element) => {
+        element.classList.remove('lang_reset');
+      });
+      document.querySelectorAll('sup').forEach((element) => {
+        element.classList.add('lang_reset');
+      });
+    } else {
+      document.querySelectorAll('span').forEach((element) => {
+        element.classList.add('lang_reset');
+      });
+      document.querySelectorAll('sup').forEach((element) => {
+        element.classList.remove('lang_reset');
+      });
+    }
+  },
+  'AltLeft',
+  'ShiftLeft',
+);
+
+
+// Save input text to localStorage
+result.value = localStorage.getItem('result');
+result.oninput = () => {
```

Comment:
always use addEvenListener for all events

Reply:
fixed